### ISSUE TITLE
WP affords moving forward and backward

### ISSUE THREAD

**Issue by iherman** [w3c/wpub#144]
User agents must provide an affordance for moving forward and backward in the default reading order of a Web Publication

**Comment by HadrienGardeur** [comment]
The more I think about this, the more I believe that this should be broken down in two separate requirements:

- a Web Publication MUST contain navigation that allows the user to read the entire reading order, without any additional affordance from the UA
- the affordance itself could become a SHOULD instead of a MUST

**Comment by iherman** [comment]
@HadrienGardeur 

> * a Web Publication MUST contain navigation that allows the user to read the entire reading order, without any additional affordance from the UA

isn't that the same as #146 ?

> * the affordance itself could become a SHOULD instead of a MUST

I do not have a problem with this; note that the text on the issue simply comes from the current draft which did not use RFC "must".

**Comment by HadrienGardeur** [comment]
> isn't that the same as #146 ?

This is about a UA affordance, I was talking about the content itself.

**Comment by iherman** [comment]
You are right, it is not the same (my mistake); in fact, in some sense it is just about the opposite unless I grossly misunderstand something.

I *think* what we're talking about is that the reader should have (to take a typical example) some ways to get to the full table of content of the publication. In a reading system like iBook or Kobo there is somewhere a button that I press and I get a window of some sort with the TOC. In those cases the navigation through the reading order is performed by the UA.

Issue #146 says (as a copy paste from the current draft):

>* User agents should provide an affordance for accessing the table of contents without leaving the resource they are currently viewing. 

which seems to reformulate the TOC handling I was talking about.

Your text says:

>* A Web Publication MUST contain navigation that allows the user to read the entire reading order, without any additional affordance from the UA

and it suggests to me that the Web Publication should carry that TOC navigation possibility with it without relying on the UA. Is this what you mean? If yes, we may have somewhat contradicting requirements; if not, I need more clarification...

**Comment by llemeurfr** [comment]
It seems to me that here again there is a blur between two notions: reading order and TOC. See my comment on the subject [here](https://github.com/w3c/wpub/issues/148#issuecomment-366942480).

**Comment by HadrienGardeur** [comment]
> and it suggests to me that the Web Publication should carry that TOC navigation possibility with it without relying on the UA. Is this what you mean? If yes, we may have somewhat contradicting requirements; if not, I need more clarification...

I meant, navigating the reading order, not the table of contents.

The example that I've used illustrates that very well: https://poignant.guide/

Every chapter has:

- top links to the next and previous resources in the reading order
- at the bottom of the page you'll also find a link to the next resource in the reading order

This makes it very easy to read the book from the first to the last resource in the reading order, without any kind of UA or affordance.

**Comment by TzviyaSiegman** [comment]
I read the topic and thought this would be about paging through a WP in the default reading order. The Default Reading Order is not necessarily the same as the TOC. Whether it is or is not, the affordance the system provides IMO is a method of navigation in addition to the TOC. This fulfills WCAG's requirement for navigating in multiple ways as well.

**Comment by iherman** [comment]
@HadrienGardeur, I understand, thanks.

However... functionally, this is equivalent to the sweeping move that one does on the border of a page in an eBook reader application on, say, my tablet. However, in those cases, it is the UA that does that. I guess what I am questioning is the MUST in your statement:

> a Web Publication MUST contain navigation that allows the user to read the entire reading order, without any additional affordance from the UA

**Comment by HadrienGardeur** [comment]
@iherman oh sure, it's equivalent.

But we want any UA to be able to read a publication from start to end, without being WP-aware.
This means that this is a MUST.

In an idea world, we also find a way to hide such links when the UA is WP-aware, but when we start such discussions we enter the realm of implementation.

**Comment by iherman** [comment]
Hm. 

On the last call @rdeltour argued that we should _not_ deal with implementation issues for now, and work out our requirements for an ideal world where the UA _is_ WP aware. (@rdeltour, I hope I got your opinion right.) And, sort of, go along with this and, for this reason, put aside the issues on how an app, mimicking a WP aware UA, would work and start up and turn into publishing mode, etc. With the idea that when we have a clear mind about the ideal world we can come back to see what we have to adapt for the less then ideal world.

However, if we go along with this line, then having a MUST is not required. It may be a SHOULD for real life...

(Note that, b.t.w., in #143, we discuss whether we should use the RFC SHOULD/MUST/etc expressions in the first place...)

**Comment by llemeurfr** [comment]
ok now that we clearly speak about next/prev navigation, I understand @HadrienGardeur 's desire to make Web Publications usable by non-WP aware UAs, therefore containing some inner navigation capabilities. 
But I'm worried that hiding such mini-navigation on WP aware UAs will be difficult, especially if the way such navigation is implemented is not severely constrained by the spec.

**Comment by HadrienGardeur** [comment]
IMO, in our affordance discussions this is the only item that is related to non-WP aware UAs.

It's fine if you don't have the other affordances with a "basic" browser, but we can't have Web Publications where you can't read the whole publication easily.

**Comment by avneeshsingh** [comment]
From accessibility perspective, it is better if publishers do not have to handle next and previous links explicitly because bdefault reading order is essential feature of WP and consistency of user interface for it is important for people with disability. i.e. people with disabilities prefer to learn a UA, and stick to it instead of learning to use each WP from scratch.

**Comment by llemeurfr** [comment]
I think @HadrienGardeur cares too much about non-WP aware UAs . 

It would be much simpler to state that: 
1/ WP-aware UAs will use the reading order integrated into the manifest to provide next/previous affordances
2/ WAM-aware UAs will be able to load a Web Application that will itself process a WP: the Web Application will afford next/previous functionalities (it's a big polyfill)
3/ for non-WP/non-WAM awares UAs, well, a provider willing to expose WPs to such old beasts will have to process WPs server-side and afford next/prev functionalities directly, in standard Web Pages (dynamically or statically generated).

I wonder if content negociation could be plugged there, but this is another thread of discussion, if this model reaches consensus.

**Comment by iherman** [comment]
> A Web Publication should not require any additional JS (polyfill, Web App...) to be read from cover to cover.

I am not sure that is achievable, though, but we should probably not try to solve this here...

**Comment by BigBlueHat** [comment]
I'm not sure providing links that afford "linear progression" across a series of resources is sufficient to make something a Web Publication.

Certainly we should provide a "progessive" pathway from the Web as it is now to the Web that we want (i.e. making non-pollyfilled WPs browsable now), but I'm not sure that needs specifying. Web authors do this now (when they want it).

However, if there are ways to leverage the presence of these links for things we need, there might be things to specify.

**Comment by avneeshsingh** [comment]
Support in existing browser is a concern. At the same time, creation of this working group means that we are also looking for WP specific functionality in the browsers. 
An approach can be to have a default expectation that UA will provide next/previous progression, and at the same time also suggest a fallback mechanism that can make it work in existing browsers.

**Comment by BigBlueHat** [comment]
Interesting history note, the original [WorldWideWeb browser](https://en.wikipedia.org/wiki/WorldWideWeb#/media/File:WorldWideWeb_FSF_GNU.png) (circa 1990 at least) had a "navigation" menu that included: `Back ^`, `Next >`, and `Previous <` options (note the lack of "forward"). Additionally, [ViolaWWW](http://viola.org/) (a few years later; 1992/1993) had matching navigation sporting the words Back, Next, and Previous along with a drop-down list of links built from `<link>` tags in the body (screenshot from [the Viola intro](http://viola.org/viola/violaIntro.html)):

> The following picture shows a "bookmark tool" that acts as a mini table of contents for the page.

![screenshot of ViolaWWW showing Previous, Back, Next, and link-tag-based Table of Contents navigation](http://viola.org/viola/violaScreenDumps/VWX-withDynaTool2_top.gif)

Seems the Web has wanted this feature since the beginning. 😄

**Comment by iherman** [comment]
Rachel will make a writeup for this affordance, based on the discussion on 2018-05-31

**Comment by RachelComerford** [comment]
Affordance name:  
WP affords moving forward and backward

1. Short description:
User agents must provide an affordance for moving forward and backward in the default reading order of a Web Publication.

2. Dependencies: 
	Moving forward and backward is dependent on the content and structure of the manifest

3. Use case Reference:
Shayna is assigned sections 2.1 through 2.5 of her textbook. She can access each section independently through links in her learning management system but prefers to read continuously without exiting the text.

Shailendra is reading about supply and demand in chapter 5 of a trade book. He recalls a helpful diagram from the end of chapter 4 and wants to return to it for a moment without navigating through all of chapter 4 to get to it.

**Comment by iherman** [comment]
@mattgarrish @RachelComerford I was wondering where in the document this would fit. For now I would think that merging that with the slightly more general [entry on reading order](https://w3c.github.io/wpub/#aff-reading-order) maybe the best fit, but then the text should be extended to the more general case.

WDYT?
